server/src/main/java/org/opensearch/indices/RemovableRBMIntKeyLookupStore.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.indices;
+
+import org.roaringbitmap.RoaringBitmap;
+import java.util.HashSet;
+
+public class RemovableRBMIntKeyLookupStore extends RBMIntKeyLookupStore implements KeyLookupStore<Integer> {
+    // The code for this class is almost the same as RemovableHybridIntKeyLookupStore,
+    // just with different superclasses.
+    // I considered changing the separate Removable classes into a CollisionHandler object
+    // which could be reused as a field of the KeyLookupStore objects, but since we will ultimately
+    // only use one of these four possible classes after doing performance testing,
+    // I don't think it's worth it to make the logic more complex just to avoid reusing code that might be deleted.
+
+    private HashSet<Integer> collidedInts;
+    private int numRemovalAttempts;
+    private int numSuccessfulRemovals;
+
```

Comment:
I think we can move these to a separate internal class KeyLookStoreStats where we hold all these. We can later extend it to have totalAdds, currentMemSize etc.
You can also take a look at CounterMetric within OpenSearch which provides a thread safe way to increment and get values. Though we are doing stuff inside lock for now.